src/vsphere_cpi/spec/integration/spec_helper.rb

```diff
@@ -15,11 +15,12 @@
 
     suite_cpi = @lifecycle_cpi
 
-    stemcell_id = ENV.fetch('BOSH_VSPHERE_STEMCELL_ID', '')
+    bosh_vsphere_stemcell_id = ENV.fetch('BOSH_VSPHERE_STEMCELL_ID', '')
 
-    unless stemcell_id.empty?
-      stemcell_vm = suite_cpi.stemcell_vm(stemcell_id)
-      fail "Could not find VM for stemcell '#{stemcell_id}'" if stemcell_vm.nil?
+    unless bosh_vsphere_stemcell_id.empty?
+      stemcell_vm = suite_cpi.stemcell_vm(bosh_vsphere_stemcell_id)
+      fail "Could not find VM for stemcell '#{bosh_vsphere_stemcell_id}'" if stemcell_vm.nil?
+      stemcell_id = bosh_vsphere_stemcell_id
```

Comment:
I cam across a bug where stemcell_id was being set to '' and when deleting stemcell it would then try to delete all vm's because regex would try to match everything. This makes sure stemcell_id is not set to empty string when BOSH_VSPHERE_STEMCELL_ID variable is not set